src/OneExplorer.ts

```diff
@@ -105,67 +107,72 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
 
-    for (const fn of files) {
-      const fpath = path.join(dirpath, fn);
+    for (const fname of files) {
+      const fpath = path.join(dirpath, fname);
       const fstat = fs.statSync(fpath);
 
       if (fstat.isDirectory()) {
         const dirNode: Node =
-            {type: NodeType.directory, name: fn, childNodes: [], uri: vscode.Uri.file(fpath)};
+            {type: NodeType.directory, name: fname, path: fpath, childNodes: [], uri: vscode.Uri.file(fpath)};
 
-        this.searchNode(dirNode, dirpath);
+        this.searchNode(dirNode);
         if (dirNode.childNodes.length > 0) {
           node.childNodes.push(dirNode);
         }
-      } else if (
-          fstat.isFile() &&
-          (fn.endsWith('.pb') || fn.endsWith('.tflite') || fn.endsWith('.onnx'))) {
-        const modelNode:
-            Node = {type: NodeType.model, name: fn, childNodes: [], uri: vscode.Uri.file(fpath)};
+      } 
+      else if 
+      (fstat.isFile() &&
+        (fname.endsWith('.pb') || fname.endsWith('.tflite') || fname.endsWith('.onnx'))) {
 
-        this.searchPairConfig(modelNode, dirpath);
+        const modelNode:
+          Node = {type: NodeType.model, name: fname, path: fpath, childNodes: [], uri: vscode.Uri.file(fpath)};
 
+        this.searchPairConfig(modelNode);
         node.childNodes.push(modelNode);
       }
     }
   }
 
   /**
    * Search .cfg files in the same directory of the node
-   *
-   * NOTE It assumes 1-1 relation for model and config
-   *
-   * TODO(dayo) Support N-N relation
-   * TODO(dayo) Search by parsing config file's model entry (Currently model name and cfg name must
-   * match)
    */
-  private searchPairConfig(node: Node, dirPath: string) {
-    const dirpath = path.dirname(path.join(dirPath, node.name));
-    const files = fs.readdirSync(dirpath);
-
-    const extSlicer = (fileName: string) => {
-      return fileName.slice(0, fileName.lastIndexOf('.'));
+  private searchPairConfig(node: Node) {
+    const modelPath = node.path;
+    const modelName = node.name;
+    const modelExt = path.extname(modelPath).slice(1);
+    
+    const readInputPath = (configPath: string) => {
+      const ini = require('ini');
+      const config = ini.parse(fs.readFileSync(configPath).toString());
+      return config[`one-import-${modelExt}`]['input_path'];
     };
 
+    const dirPath = path.dirname(modelPath);
+    const files = fs.readdirSync(dirPath);
     for (const fn of files) {
-      const fpath = path.join(dirpath, fn);
+      const fpath = path.join(dirPath, fn);
       const fstat = fs.statSync(fpath);
 
-      if (fstat.isFile() && fn.endsWith('.cfg') && (extSlicer(fn) === extSlicer(node.name))) {
-        const configNode:
-            Node = {type: NodeType.config, name: fn, childNodes: [], uri: vscode.Uri.file(fpath)};
-        node.childNodes.push(configNode);
+      if (fstat.isFile() && fn.endsWith('.cfg')) {
+        if (readInputPath(fpath) === modelName) {
```

Comment:
Assuming model name is `"foo.tflite"`,
I guess `readInputPath(fpath)` could return 
1. `"foo.tflite"`
2. `"./foo.tflite"`
3. `"/home/eric/model/inception/foo.tflite"`

maybe its better check basename of return value from `readInputPath(fpath)`.

Reply:
Sure :-D